lua/cfc_adverts/server/sv_adverts.lua

```diff
@@ -0,0 +1,77 @@
+CFCAdverts = CFCAdverts or {}
+
+CFCAdverts._advertCount = 0
+CFCAdverts._curAdvert = 1
+CFCAdverts._advertKeys = {}
+CFCAdverts._timerName = CFCAdverts.HOOK_BASE .. "_SendAdvert"
+
+local function buildNotif( notif, advert )
+    notif:SetTitle( advert.title )
+    notif:SetText( advert.text )
+    notif:SetTextColor( advert.textColor )
+    notif:SetDisplayTime( advert.displayTime )
+    notif:SetPriority( advert.priority )
+    notif:SetCloseable( advert.closeable )
+    notif:SetIgnoreable( advert.ignoreable )
+    notif:SetTimed( advert.timed )
+
+    if advert.type == "Buttons" then
+        for i, button in pairs( advert.buttons ) do
+            notif:AddButton( button.text, button.buttonColor, i )
+        end
+
+        function notif:OnButtonPressed( ply, index, ... )
+            local button = advert.buttons[index]
+
+            button.func( ply, arg )
+
+            if not button.closeOnPress then return end
+
+            notif:RemovePopup( notif:GetCallingPopupID(), ply )
+        end
+    end
+end
+
+function CFCAdverts.updateAdverts()
+    CFCAdverts._advertCount = table.Count( CFCAdverts.adverts )
+    CFCAdverts._advertKeys = table.GetKeys( CFCAdverts.adverts )
+
+    if CFCAdverts._advertCount == 0 then
+        timer.Simple( 30, CFCAdverts.updateAdverts ) -- Prevent CFC Adverts from dying forever if the list goes empty
+        return
+    end
+
+    CFCAdverts._curAdvert = math.min( CFCAdverts._curAdvert, CFCAdverts._advertCount )
+    local advertGap = CFCAdverts.cycleTime / CFCAdverts._advertCount
+
+    timer.Create( CFCAdverts._timerName, advertGap, 0, function()
+        local plys = player.GetHumans()
+
+        if table.IsEmpty( plys ) then return end
+
+        local advertName = CFCAdverts._advertKeys[CFCAdverts._curAdvert] or ""
+        local advert = CFCAdverts.adverts[advertName] or {}
+
+        if table.IsEmpty( advert ) or table.Count( CFCAdverts.adverts ) ~= CFCAdverts._advertCount then
```

Comment:
I think it'd be nicer to just do this
```suggestion
        local advert = CFCAdverts.adverts[advertName]

        if not advert or table.Count( CFCAdverts.adverts ) ~= CFCAdverts._advertCount then
```